### PR TITLE
Add guidelines for becoming a "core developer" of Castro

### PR DESCRIPTION
**Pull request by zingale**
This helps to formalize the process for recognition for contributors to Castro.

### CODE REVIEW

**Comment by maxpkatz** [comment]
Can you add a comment to the CHANGES file briefly describing this as part of the PR?

File: README.md [hunk 1]
```diff
@@ -39,19 +39,69 @@ http://bender.astro.sunysb.edu/Castro/staging/Castro/html/
...
+    Castro and target the `development` branch.  (If you mistakenly
+    target `master`, we can change it for you.)
+
+    Please add a line to `CHANGES` summarizing your change if it
+    is a bug fix or new feature.  Reference the PR or issue as
+    appropriate
+
+    If there are a number of small commits making up the PR, we may
+    wish to squash commits upon merge to have a clean history.
+    *Please ensure that your PR title and first post are descriptive,
```
Comment (cmsquared) [inline README.md]: "First post" here means the PR description, correct? Also, this isn't a requirement of squashing...but I assume it is what comes up when you click on "Squash and merge" as the default?

**Comment by cmsquared** [comment]
You already merged it, bu this seems reasonable to me.

**Comment by zingale** [comment]
yeah, it seems squash and merge does a pretty good job

**Pull request merged**